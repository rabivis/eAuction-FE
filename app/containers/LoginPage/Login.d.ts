export interface LoginFormParams {
    userName: string;
    userPass: string;
}

export interface CurrentUser {
    user: User;
    token: string;
}

export interface User{
    id: number;
    firstName: string;
    lastName: string;
    email: string;
    address: string;
    city: string;
    state: string;
    phone: string;
    password: string;
    pin: string;
    userType: number;
}