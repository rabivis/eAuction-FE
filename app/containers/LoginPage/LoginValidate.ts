import { FormErrors } from 'redux-form';
import { LoginFormParams } from './Login.d';

const LoginFormvalidate = (values: LoginFormParams): FormErrors<LoginFormParams> => {
    const errors: FormErrors<LoginFormParams> = {};

    if (!values.userName) {
      errors.userName = 'User name required';
    }

    if (!values.userPass) {
      errors.userPass = 'Password required';
    }
    return errors;
};
export default LoginFormvalidate;