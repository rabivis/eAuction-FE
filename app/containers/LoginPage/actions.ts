import { action } from 'typesafe-actions';
import { LoginFormParams, CurrentUser } from './Login.d';

import ActionTypes from './constants';

export const loginUserSubmit = (formData: LoginFormParams) =>
    action(ActionTypes.LOGIN_USER, { formData } );

export const updateCurrentUser = (currentUser: CurrentUser) =>
    action(ActionTypes.LOGIN_USER_SUCCESS, { currentUser });

export const loginError = (error: any) =>
    action(ActionTypes.LOGIN_USER_ERROR, error);