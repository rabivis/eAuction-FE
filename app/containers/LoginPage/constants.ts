
enum ActionTypes {
    LOGIN_USER = 'eAucation/LOGIN_USER',
    LOGIN_USER_SUCCESS = 'eAucation/LOGIN_USER_SUCCESS',
    LOGIN_USER_ERROR = 'eAucation/LOGIN_USER_ERROR',
  }
  
  export default ActionTypes;
  