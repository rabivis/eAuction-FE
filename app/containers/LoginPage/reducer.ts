import { ContainerState, ContainerActions } from './types';
import ActionTypes from './constants';

// The initial state of the App
export const initialState: ContainerState = {
  loading: false,
  error: false,
  currentUser: undefined,
};

// Take this container's state (as a slice of root state), this container's actions and return new state
function LoginReducer(
  state: ContainerState = initialState,
  action: ContainerActions,
): ContainerState {
  switch (action.type) {
    case ActionTypes.LOGIN_USER:
      return {
        loginFormParams: action.payload.formData,
        loading: true,
        error: false,
      };
    case ActionTypes.LOGIN_USER_SUCCESS:
      return {
        currentUser: action.payload.currentUser,
        loading: false,
        error: state.error,
      };
    case ActionTypes.LOGIN_USER_ERROR:
      const { error, loading, ...rest } = state;
      return {
        error: action.payload,
        loading: false,
        ...rest,
      };
    default:
      return state;
  }
}

export default LoginReducer;
