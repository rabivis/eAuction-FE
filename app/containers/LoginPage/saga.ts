/**
 * Gets the repositories of the user from Github
 */

import { call, put, select, takeLatest } from 'redux-saga/effects';
import { updateCurrentUser, loginError } from './actions';
import ActionTypes from './constants';

import request from 'utils/request';

/**
 * Github repos request/response handler
 */
export function* loginUser(action) {
  const { userName, userPass } = action.payload.formData;
  const requestURL = `http://localhost:8090/api/v1/user/login`;
  try {
    // Call our request helper (see 'utils/request')
    const userData = yield call(request, requestURL, {
      method: 'POST',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({ email: userName, password: userPass })
    });
    
    yield put(updateCurrentUser({user: userData.user, token: userData.token}));
  } catch (err) {
    yield put(loginError(err));
  }
}

/**
 * Root saga manages watcher lifecycle
 */
export default function* connectToBackend() {
  // Watches for LOAD_REPOS actions and calls getRepos when one comes in.
  // By using `takeLatest` only the result of the latest API call is applied.
  // It returns task descriptor (just like fork) so we can continue execution
  // It will be cancelled automatically on component unmount
  yield takeLatest(ActionTypes.LOGIN_USER, loginUser);
}
