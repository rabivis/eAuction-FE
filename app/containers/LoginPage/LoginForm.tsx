import React, { useEffect } from 'react';
import { Form, FormGroup, Button, Row, Col } from "reactstrap";
import { Field, reduxForm } from 'redux-form';
import { connect } from 'react-redux';
import LoginFormvalidate from './LoginValidate'
import ReduxFormInput from 'components/Inputs/ReduxFormInput';
import { LoginFormParams } from './Login.d';
import { loginUserSubmit } from './actions';
import { useDispatch, useSelector } from 'react-redux';
import { useInjectReducer, useInjectSaga } from 'utils/redux-injectors';
import { createStructuredSelector } from 'reselect';
import { makeSelectCurrentUser, makeSelectLoading, makeSelectError } from './selectors';
import { useHistory } from 'react-router-dom';

import reducer from './reducer';
import saga from './saga';

interface Props {
    handleSubmit;
    pristine;
    submitting;
    isLoading;
};

const key = "login";


const stateSelector = createStructuredSelector({
    currentUser: makeSelectCurrentUser(),
    loading: makeSelectLoading(),
    error: makeSelectError(),
});

export function LoginForm(props: Props) {

    const dispatch = useDispatch();
    const { currentUser, loading, error } = useSelector(stateSelector);
    const history = useHistory();
    if (currentUser != undefined) {
        history.push("/");
    }
    
    const { handleSubmit, submitting, pristine, isLoading } = props;
    const localSubmithandle = (values: LoginFormParams) => {
        dispatch(loginUserSubmit(values))
    }

    useInjectReducer({ key: key, reducer: reducer });
    useInjectSaga({ key: key, saga: saga });

    return (
        <Form onSubmit={handleSubmit(localSubmithandle)} noValidate={true}>
            <Field
                name="userName"
                type="text"
                component={ReduxFormInput}
                label="User Name *"
                placeHolder="Enter User Name"

            />
            <Field
                name="userPass"
                type="password"
                component={ReduxFormInput}
                label="Password *"
                placeHolder="Enter Password"

            />
            <Row>
                <Col md="12">
                    <Button
                        className="btn btn-success placeicon"
                        color="success"
                        type="submit"
                        disabled={pristine || submitting}
                    >
                        {isLoading && <>Login...</>}
                        {!isLoading && <>Login</>}
                    </Button>
                </Col>
            </Row>
        </Form>
    );
}

const form = reduxForm<{}, Props>({
    destroyOnUnmount: false,
    forceUnregisterOnUnmount: false,
    form: 'loginForm',
    validate: LoginFormvalidate
})(LoginForm);

export default connect(null)(form);


