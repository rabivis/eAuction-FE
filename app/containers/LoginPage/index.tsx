import React from 'react';
import { Helmet } from 'react-helmet-async';
import LoginForm from './LoginForm';
import Helpdesk from 'components/Helpdesk';
import { createStructuredSelector } from 'reselect';
import { makeSelectError } from './selectors';
import { useSelector } from 'react-redux';

const stateSelector = createStructuredSelector({
    error: makeSelectError(),
});

export default function LoginPage() {

    const { error } = useSelector(stateSelector);
    return (
        <>
            <Helmet>
                <title>Sign In</title>
                <meta
                    name="description"
                    content="Login - eAuction"
                />
            </Helmet>
            <Helpdesk 
                message={(error)?'Invalid user name & password':false}
                messageType={ (error)? 'danger': ''}
            />
            <div className="row justify-content-center">
                <div className="text-center" id="test">Sign In</div>
            </div>
            <div className="row justify-content-center">
                <div className="col-11">
                    <div className="form-card">
                        <LoginForm />
                    </div>
                </div>
            </div>
        </>
    );
}