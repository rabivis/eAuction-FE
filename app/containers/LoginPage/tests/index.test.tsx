import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import 'jest-styled-components';

import LoginPage from '../index';

const renderer = createRenderer();

describe('<LoginPage />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<LoginPage />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
