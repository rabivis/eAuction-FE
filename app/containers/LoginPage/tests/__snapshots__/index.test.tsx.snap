// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should render and match the snapshot 1`] = `
<React.Fragment>
  <Helmet
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Sign In
    </title>
    <meta
      content="Login form for login in Bank Management System"
      name="description"
    />
  </Helmet>
  <Helpdesk />
  <div
    className="row justify-content-center"
  >
    <div
      className="text-center"
      id="test"
    >
      Sign In
    </div>
  </div>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-11"
    >
      <div
        className="form-card"
      >
        <Memo(Connect(ReduxForm)) />
      </div>
    </div>
  </div>
</React.Fragment>
`;
